### PR TITLE
Add test invalid signature to wallet

### PR DESCRIPTION
**Pull request by ID25**

### CODE REVIEW

File: lib/zilliqa/account/wallet.rb [hunk 1]
```diff
@@ -135,10 +146,24 @@ def sign_with(tx, address)
         end
 
         tx.sender_pub_key = account.public_key
-        sig = account.sign_transaction(tx)
+
+        begin
+          sig ||= account.sign_transaction(tx)
+          raise InvalidSignatureSizeError.new(sig.to_s) unless valid_signature?(sig)
```
Comment (vp993) [inline lib/zilliqa/account/wallet.rb]: Wouldn't `if` statement here be more effective? `raise` & `rescue` are very resource consuming.

File: test/account/wallet_test.rb [hunk 2]
```diff
@@ -68,4 +68,47 @@ def test_sign
...
+        gas_price: rand(10..15_000),
+        gas_limit: '1',
+        to_addr: '1234567890123456789012345678901234567890',
+        code: TEST_CONTRACT.gsub('/\\', ''),
+        data: JSON.generate(ABI)
+      }
+
+      tx = Zilliqa::Account::Transaction.new(tx_params, nil)
+
+      wallet.sign(tx)
```
Comment (vp993) [inline test/account/wallet_test.rb]: Could you add sign validating here to add a sense for the test?

**Review by vp993 — CHANGES_REQUESTED** [review]

File: lib/zilliqa/account/wallet.rb (hunk 1 above)
Comment (vp993) [reply lib/zilliqa/account/wallet.rb]: Put it into Schnorr class

File: test/account/wallet_test.rb [hunk 3]
```diff
@@ -68,4 +68,48 @@ def test_sign
...
+        gas_price: rand(10..15_000),
+        gas_limit: '1',
+        to_addr: '1234567890123456789012345678901234567890',
+        code: TEST_CONTRACT.gsub('/\\', ''),
+        data: JSON.generate(ABI)
+      }
+
+      tx = Zilliqa::Account::Transaction.new(tx_params, nil)
+      sig = account.sign_transaction(tx)
+      Zilliqa::Util::Validator.signature?(sig.to_s)
```
Comment (vp993) [inline test/account/wallet_test.rb]: Is assert or something alike missed here?

Comment (ID25) [reply test/account/wallet_test.rb]: For bool result it not necessary, but i add this for read readability

**Review by vp993 — APPROVED** [review]

**Pull request merged**